manifests/puppetserver.pp

```diff
@@ -17,5 +17,31 @@
     hosts         => $hosts,
     metrics_port  => $port,
   }
+  
+  if $facts['pe_server_version'] < '2018.1.0' {
+  
+    Pe_metric_curl_cron_jobs::Pe_metric <| title == 'puppetserver' |> {
+      additional_metrics => [
```

Comment:
Couple of small changes:

  - When comparing version numbers, we should use the `versioncmp()` function as string comparison does odd things like treating `10` as being less than `2`.

  - Using a collector override isn't necessary in this instance as we can pass the additional metrics directly to the `puppet_metrics_collector::pe_metric` declaration on line 15.

Something like the following:

```puppet
if versioncmp($facts['pe_server_version'], '2018.1.0') < 0 {
  $additional_metrics = [
    ...
  ]
} else {
  $additional_metrics = []
}

puppet_metrics_collector::pe_metric { 'puppetserver' :
  ...,
  additional_metrics => $additional_metrics,
}
```